### PR TITLE
docs: Update for React 19 compatible info

### LINKED ISSUE #52106: 官方示例中的confirm with promise 示例不work
### Reproduction link

[https://ant.design/components/modal-cn#modal-demo-confirm](https://ant.design/components/modal-cn#modal-demo-confirm)

### Steps to reproduce

点击“with promise” 按钮后，点击“确定”按钮

### What is expected?

点击“with promise” 按钮后，点击“确定”按钮，弹窗关闭

### What is actually happening?

点击“with promise” 按钮后，点击“确定”按钮，弹窗未关闭

| Environment | Info |
| --- | --- |
| antd | 5.22.6 |
| React | 18 |
| System | macOs 14.5 |
| Browser | Chrome Version 131.0.6778.205 |

### PR DESCRIPTION
**Pull request by zombieJ**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [ ] 🐞 Bug fix
- [x] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

fix #52106

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     -      |
| 🇨🇳 Chinese |     -      |

### CODE REVIEW

File: docs/react/v5-for-19.zh-CN.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+<!-- prettier-ignore -->
+:::info{title="兼容接口"}
+antd v5 默认兼容 React 16 ~ 18 版本，对于 React 19 版本，可以使用以下兼容方法进行适配。该兼容方式以及接口将在 v6 被移除。
+:::
+
+### React 19 兼容问题
+
+由于 React 19 调整了 `react-dom` 的导出方式，导致 antd 无法直接使用 `ReactDOM.render` 方法。因而使用 antd 会遇到以下问题：
+
+- 波纹特效无法正常工作（如 Button、Switch 等组件点击控制台报错）
```
Comment (afc163) [inline docs/react/v5-for-19.zh-CN.md:18]: > 如 Button、Switch 等组件点击控制台报错

现在不会报错。

File: docs/react/v5-for-19.zh-CN.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+- `Modal`、`Notification`、`Message` 等组件的静态方法报错
```
Comment (afc163) [inline docs/react/v5-for-19.zh-CN.md:19]: 可以写一下，useApp、useXxx 等调用方式不会报错。

File: docs/react/v5-for-19.zh-CN.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+unstableSetRender((node, container) => {
+  container._reactRoot ||= createRoot(container);
+  const root = container._reactRoot;
+  root.render(node);
+  return async () => {
+    await new Promise((resolve) => setTimeout(resolve, 0));
+    root.unmount();
+  };
+});
+```
```
Comment (afc163) [inline docs/react/v5-for-19.zh-CN.md:58]: 这一段不用写了吧。

File: .dumi/theme/layouts/GlobalLayout.tsx [hunk 4]
```diff
@@ -23,6 +22,8 @@ import SiteThemeProvider from '../SiteThemeProvider';
 import type { SiteContextProps } from '../slots/SiteContext';
 import SiteContext from '../slots/SiteContext';
 
+import '@ant-design/v5-patch-for-react-19';
```
Comment (afc163) [inline .dumi/theme/layouts/GlobalLayout.tsx:25]: 我看这么简单一个包编译完还引入了 regeneratorRuntime https://unpkg.com/browse/@ant-design/v5-patch-for-react-19@1.0.1/lib/index.js ，会不会没必要。源码直接写 Promise 写法不用 async / await 是不是就够了。

File: docs/react/v5-for-19.zh-CN.md (hunk 3 above)
Comment (zombieJ) [reply docs/react/v5-for-19.zh-CN.md:58]: > 除非对于 umd、微应用等特殊场景，才使用 `unstableSetRender` 方法。

多 antd 实例可能会对不到 `v5`，里面也推荐优先用包了。

**Review by li-jia-nan — APPROVED** [review]

File: docs/react/v5-for-19.zh-CN.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  title: 进阶使用
+order: 9
+title: React 19 兼容
+tag: New
+---
+
+<!-- prettier-ignore -->
+:::info{title="兼容接口"}
+antd v5 默认兼容 React 16 ~ 18 版本，对于 React 19 版本，可以使用以下兼容方法进行适配。该兼容方式以及接口将在 v6 被移除。
+:::
```
Comment (afc163) [inline docs/react/v5-for-19.zh-CN.md:12]: @zombieJ 需要写一下从哪个版本开始兼容 React 19

changelog 里的对应版本也标记一下。

**Pull request merged**